indexing-service/src/test/java/org/apache/druid/indexing/overlord/ForkingTaskRunnerTest.java

```diff
@@ -563,4 +576,57 @@ int waitForTaskProcessToComplete(Task task, ProcessHolder processHolder, File lo
     Assert.assertEquals(1L, (long) forkingTaskRunner.getWorkerFailedTaskCount());
     Assert.assertEquals(0L, (long) forkingTaskRunner.getWorkerSuccessfulTaskCount());
   }
+
+  @Test
+  public void testCannotRestoreTasks() throws Exception
```

Comment:
doubt : does this test verify that a cleanly finished task is removed from the restore list?

Reply:
This test verifies that restore fails when a task is in the list of tasks to be restored but has no previous working directory